### DISCUSSION TITLE
Working with canvas

### DISCUSSION THREAD

**Discussion by TheNtwork** [capricorn86/happy-dom#1056] [General]
Happy-Dom implements a lot of the normal Document features in regards to reading from the parsed html such as getElementById etc.
However i couldnt find any documentation in regards to tag specific functions such as document.createElement("canvas").getContext

Is there any particular file where adding this functionality is possible (ex. a handler that determines wich properties of the dom get exposed in regards to certain calls) I couldnt find anything when looking through the source but i might be mistaken

**Comment by capricorn86** [comment]
Hi @TheNtwork! :slightly_smiling_face: 

Happy DOM has not implemented Canvas yet unfortunately. It is tracked in the #241 task.

**Reply by TheNtwork** [reply]
Im reffering to that as an example
Sorry if my question was unclear i was asking if there was some handle or something in place that determines wich element has wich dom features so that i can potentially add this and other functionality myself

**Reply by capricorn86** [reply]
There is no document/page that tracks which elements has implemented which features yet unfortunately.

You can find all element tags that has an assigned element class for them here:
https://github.com/capricorn86/happy-dom/blob/master/packages/happy-dom/src/config/ElementTag.ts

Here you will find a list of all elements that has not been implemented yet here:
https://github.com/capricorn86/happy-dom/blob/master/packages/happy-dom/src/config/NonImplemenetedElementClasses.ts

Note that many of the elements doesn't need any specific implementation class to work as expected as they don't have any special functionality other than what is inherited from HTMLElement.

If you want to add your own implementation you could do it like this:
```javascript
import { HTMLElement } from 'happy-dom';
import ElementTag from 'happy-dom/{lib or cjs}/config/ElementTag.{js or cjs}';

class HTMLCanvasElement extends HTMLElement {
        // Do implementation
}

ElementTag['CANVAS'] = HTMLCanvasElement;
```
